src/hera/workflow.py

```diff
@@ -320,3 +328,12 @@ def get_parameter(self, name: str) -> Parameter:
         if self.parameters is None or next((p for p in self.parameters if p.name == name), None) is None:
             raise KeyError(f"`{name}` is not a valid workflow parameter")
         return Parameter(name, value=f"{{{{workflow.parameters.{name}}}}}")
+
+    def to_dict(self) -> dict:
```

Comment:
Should this take an optional `serialize` kwarg?

Reply:
The `to_dict` method on the built object does not accept the serialize, but the internal call to the `argo_workflows` package serializer does, so we can expose that as an optional parameter through that call if any users would like to use that

Reply:
I noticed the suggestion to add a `to_json`, which I quite like, so will skip adding the serialize kwarg and instead expose the `to_json` method